File: recovered-source/jadx/sources/me/ag2s/epublib/domain/TOCReference.java
package me.ag2s.epublib.domain;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

/* JADX INFO: loaded from: app-classes.jar:me/ag2s/epublib/domain/TOCReference.class */
public class TOCReference extends TitledResourceReference implements Serializable {
    private static final long serialVersionUID = 5787958246077042456L;
    private List<TOCReference> children;
    private static final Comparator<TOCReference> COMPARATOR_BY_TITLE_IGNORE_CASE = (tocReference1, tocReference2) -> {
        return String.CASE_INSENSITIVE_ORDER.compare(tocReference1.getTitle(), tocReference2.getTitle());
    };

    @Deprecated
    public TOCReference() {
        this(null, null, null);
    }

    public TOCReference(String name, Resource resource) {
        this(name, resource, null);
    }

    public TOCReference(String name, Resource resource, String fragmentId) {
        this(name, resource, fragmentId, new ArrayList());
    }

    public TOCReference(String title, Resource resource, String fragmentId, List<TOCReference> children) {
        super(resource, title, fragmentId);
        this.children = children;
    }

    public static Comparator<TOCReference> getComparatorByTitleIgnoreCase() {
        return COMPARATOR_BY_TITLE_IGNORE_CASE;
    }

    public List<TOCReference> getChildren() {
        return this.children;
    }

    public TOCReference addChildSection(TOCReference childSection) {
        this.children.add(childSection);
        return childSection;
    }

    public void setChildren(List<TOCReference> children) {
        this.children = children;
    }
}

File: src/main/kotlin/me/ag2s/epublib/domain/TOCReference.kt
package me.ag2s.epublib.domain

import java.io.Serializable
import java.util.Comparator

class TOCReference(title: String?, resource: Resource?, fragmentId: String?, private var children: MutableList<TOCReference>) : TitledResourceReference(resource, title, fragmentId), Serializable {
    @Deprecated("Deprecated in original API") constructor() : this(null, null, null)
    constructor(name: String?, resource: Resource?) : this(name, resource, null)
    constructor(name: String?, resource: Resource?, fragmentId: String?) : this(name, resource, fragmentId, ArrayList())
    fun getChildren(): MutableList<TOCReference> = children
    fun addChildSection(childSection: TOCReference): TOCReference { children.add(childSection); return childSection }
    fun setChildren(children: MutableList<TOCReference>) { this.children = children }
    companion object {
        private const val serialVersionUID = 5787958246077042456L
        private val comparatorByTitleIgnoreCase = Comparator<TOCReference> { first, second -> String.CASE_INSENSITIVE_ORDER.compare(first.getTitle(), second.getTitle()) }
        @JvmStatic fun getComparatorByTitleIgnoreCase(): Comparator<TOCReference> = comparatorByTitleIgnoreCase
    }
}
